tests/test_vms_exist.py

```diff
@@ -32,6 +33,10 @@ def test_expected(self):
         expected_vms = set(SD_VMS + SD_DVM_TEMPLATES + SD_TEMPLATES)
         self.assertEqual(set(sdw_tagged_vm_names), set(expected_vms))
 
+        # Check for untagged VMs
+        for vm_name in SD_UNTAGGED_DEPRECATED_VMS:
+            self.assertRaises(KeyError, lambda: self.app.domains[vm_name])  # noqa: PT027
```

Comment:
A bit of a weird construct, could've been `self.assertNotIn(vm_name, self.app.domains)`?